AcadeMeData/tests.py

```diff
@@ -47,51 +35,116 @@ def test_get_user(self):
-                                                   description="A cool guy who looked familiar",
-                                                   rate=4.5)
-        assert professor_for_example.get_name() == "DR Arnold Schwarzenegger"
-
-    def test_create_professor(self, name="DR Arnold Schwarteneiger"):
-        professor_for_example = self.get_proffesor()
-        professor = Professor.get_proffesor(name)
+    def test_get_name(self, professor_id=1, name="DR Arnold Schwarzenegger", university=None,
+                      description="A cool guy who looked familliar", rate=4.5, university_id=1,
+                      uni_name='Reichman University', location="Herzlia",
+                      uni_description="A nice place"):
+        university = TestUniversityModel.generate_university(self, university_id, uni_name, location, uni_description)
+        professor_for_example = self.generate_professor(professor_id, name, university, description, rate)
+        assert professor_for_example.get_name() == name
+
+    def test_create_professor(self, professor_id=1, name="DR Arnold Schwarzenegger", university=None,
+                              description="A cool guy who looked familliar", rate=4.5, university_id=1,
+                              uni_name='Reichman University', location="Herzlia",
+                              uni_description="A nice place"):
+        university = TestUniversityModel.generate_university(self, university_id, uni_name, location, uni_description)
+        professor_for_example = self.generate_professor(professor_id, name, university, description, rate)
+        professor = Professor.get_professor(name)
         assert professor.get_name() == professor_for_example.name
         assert professor.get_description() == professor_for_example.description
+
+
+@pytest.mark.django_db
+class TestMessageBoardModel:
+    @pytest.fixture
+    def generate_msgboard(self, id=1, courseName="Linear Algebra"):
+        msgboard = MessageBoards(id=id, courseName=courseName)
+        msgboard.save()
+        return msgboard
+
+    def test_get_msgboard(self, generate_msgboard):
+        msgboard = generate_msgboard
+        msgboard_test = MessageBoards.get_msgboard_by_id(1)
+        assert msgboard_test == msgboard
+        assert isinstance(msgboard_test, MessageBoards)
+
+
+@pytest.mark.django_db
+class TestMessagesModel:
+    @pytest.fixture
+    def user_example(self):
+        user_data = {'username': "username2212", 'password': "password", 'email': "user@example.com", 'type': "S",
+                     'university': "RU",
+                     'degree': "CS"}
+        user = User.create_user(*user_data)
+        return user
+
+    def generate_message(self, msgID=1, userID=None, text='This is a test message yo'):
+        message = Messages(msgID=msgID, userID=userID, text=text)
+        message.save()
+        return message
+
+    def test_get_msg(self, user_example, msgID=1, userID=None, text='This is a test message yo'):

```

Comment:
Call fixture here

Reply:
I do call it no? see user_example is in the arguments.